ethToElrond/v2/ethToElrondBridgeExecutor_test.go

```diff
@@ -0,0 +1,601 @@
+	}
+}
+
+func TestNewEthToElrondBridgeExecutor(t *testing.T) {
+	t.Parallel()
+
+	t.Run("nil logger should error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockEthToElrondExecutorArgs()
+		args.Log = nil
+		executor, err := NewEthToElrondBridgeExecutor(args)
+
+		assert.True(t, check.IfNil(executor))
+		assert.Equal(t, errNilLogger, err)
+	})
+	t.Run("nil elrond client should error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockEthToElrondExecutorArgs()
+		args.ElrondClient = nil
+		executor, err := NewEthToElrondBridgeExecutor(args)
+
+		assert.True(t, check.IfNil(executor))
+		assert.Equal(t, errNilElrondClient, err)
+	})
+	t.Run("nil ethereum client should error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockEthToElrondExecutorArgs()
+		args.EthereumClient = nil
+		executor, err := NewEthToElrondBridgeExecutor(args)
+
+		assert.True(t, check.IfNil(executor))
+		assert.Equal(t, errNilEthereumClient, err)
+	})
+	t.Run("nil topology provider should error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockEthToElrondExecutorArgs()
+		args.TopologyProvider = nil
+		executor, err := NewEthToElrondBridgeExecutor(args)
+
+		assert.True(t, check.IfNil(executor))
+		assert.Equal(t, errNilTopologyProvider, err)
+	})
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockEthToElrondExecutorArgs()
+		executor, err := NewEthToElrondBridgeExecutor(args)
+
+		assert.False(t, check.IfNil(executor))
+		assert.Nil(t, err)
+	})
+}
+
+func TestEthToElrondBridgeExecutor_GetLogger(t *testing.T) {
+	t.Parallel()
+
+	args := createMockEthToElrondExecutorArgs()
+	executor, _ := NewEthToElrondBridgeExecutor(args)
+
+	assert.True(t, args.Log == executor.GetLogger()) //pointer testing
+}
+
+func TestEthToElrondBridgeExecutor_MyTurnAsLeader(t *testing.T) {
+	t.Parallel()
+
+	args := createMockEthToElrondExecutorArgs()
+	wasCalled := false
+	args.TopologyProvider = &bridgeV2.TopologyProviderStub{
+		MyTurnAsLeaderCalled: func() bool {
+			wasCalled = true
+			return true
+		},
+	}
+
+	executor, _ := NewEthToElrondBridgeExecutor(args)
+	assert.True(t, executor.MyTurnAsLeader())
+	assert.True(t, wasCalled)
+}
+
+func TestEthToElrondBridgeExecutor_GetAndStoreActionID(t *testing.T) {
+	t.Parallel()
+
+	t.Run("nil batch should error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockEthToElrondExecutorArgs()
+		executor, _ := NewEthToElrondBridgeExecutor(args)
+
+		actionID, err := executor.GetAndStoreActionID(context.Background())
+		assert.Zero(t, actionID)
+		assert.Equal(t, errNilBatch, err)
+	})
+	t.Run("elrond client errors", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockEthToElrondExecutorArgs()
+		expectedErr := errors.New("expected error")
+		providedBatch := &clients.TransferBatch{}
+
+		args.ElrondClient = &bridgeV2.ElrondClientStub{
+			GetActionIDForProposeTransferCalled: func(ctx context.Context, batch *clients.TransferBatch) (uint64, error) {
```

Comment:
As discussed in the past with @bogdan-rosianu , I wouldn't change the anonymous function arguments for the following reasons:
1. it seems it is not a weak warning (yet) for an anonymous function to have the unused parameters named as `_`
2. if we keep the naming for the parameters, we will provide information in case we need to use that parameter if we need to change the code. Otherwise we will need to go back and forth to figure out how to consistently name the parameter and what it represents. This is also valid when defining interfaces. For information purposes I've chosen to always name the parameters in an interface definition

Reply:
Right, good point and explanation. :+1: